cmd/soroban-rpc/internal/db/transaction.go

```diff
@@ -0,0 +1,387 @@
+	reader, err := txn.NewTx(ctx)
+	defer reader.Done()
+	if err != nil {
+		return ledgerbucketwindow.LedgerRange{}
+	}
+
+	lRange := reader.GetLedgerRange()
+	return lRange
+}
+
+func (txn *transactionReaderTx) Done() error {
+	return txn.db.Rollback()
+}
+
+// GetLedgerRange pulls the min/max ledger sequence numbers from the database.
+func (txn *transactionReaderTx) GetLedgerRange() ledgerbucketwindow.LedgerRange {
+	var ledgerRange ledgerbucketwindow.LedgerRange
+	log.Debugf("Retrieving ledger range from database: %+v", txn)
+
+	sqlTx := txn.db.GetTx()
+	newestQ := sq.Select("meta").
+		From(ledgerCloseMetaTableName).
+		OrderBy("sequence DESC").
+		Limit(1).
+		RunWith(sqlTx)
+	rows, err := newestQ.Query()
+	if err != nil {
+		log.Errorf("Error when querying for latest ledger: %v", err)
+		return ledgerRange
+	}
+
+	// There is almost certainly a row, but we want to avoid a race condition
+	// with ingestion as well as support test cases from an empty DB, so we need
+	// to sanity check that there is in fact a result.
+	if !rows.Next() {
+		if ierr := rows.Err(); ierr != nil {
+			log.Errorf("Error when querying for latest ledger: %v", ierr)
+		}
+		return ledgerRange
+	}
+
+	var row1, row2 []byte
+	var lcm1, lcm2 xdr.LedgerCloseMeta
+	if err := rows.Scan(&row1); err != nil {
+		log.Errorf("Error when scanning newest ledger: %v", err)
+		return ledgerRange
+	}
+	if err := lcm1.UnmarshalBinary(row1); err != nil {
+		log.Errorf("Error when unmarshaling newest ledger: %v", err)
+		return ledgerRange
+	}
+	// Best effort: if the second fails later, we at least set the first.
+	ledgerRange.FirstLedger.Sequence = lcm1.LedgerSequence()
+	ledgerRange.FirstLedger.CloseTime = lcm1.LedgerCloseTime()
+	ledgerRange.LastLedger.Sequence = lcm1.LedgerSequence()
+	ledgerRange.LastLedger.CloseTime = lcm1.LedgerCloseTime()
+
+	oldestQ := sq.Select("meta").
+		From(ledgerCloseMetaTableName).
+		OrderBy("sequence ASC").
+		Limit(1).
+		RunWith(sqlTx).
+		QueryRow()
+
+	if err := oldestQ.Scan(&row2); err != nil {
+		log.Errorf("Error when scanning oldest ledger: %v", err)
+		return ledgerRange
+	}
+	if err := lcm2.UnmarshalBinary(row2); err != nil {
+		log.Errorf("Error when unmarshaling oldest ledger: %v", err)
+		return ledgerRange
+	}
+	ledgerRange.FirstLedger.Sequence = lcm2.LedgerSequence()
+	ledgerRange.FirstLedger.CloseTime = lcm2.LedgerCloseTime()
+
+	log.Debugf("Database ledger range: [%d, %d]",
+		ledgerRange.FirstLedger.Sequence, ledgerRange.LastLedger.Sequence)
+	return ledgerRange
+}
+
+// GetTransaction conforms to the interface in
+// methods/get_transaction.go#NewGetTransactionHandler so that it can be used
+// directly against the RPC handler.
+//
+// Errors (i.e. the bool being false) should only occur if the XDR is out of
+// date or is otherwise incorrect/corrupted or the tx really isn't in the DB.
+func (txn *transactionReaderTx) GetTransaction(log *log.Entry, hash xdr.Hash) (
```

Comment:
instead of taking a logger parameter, this function should return an error and then the caller can decide how to handle the error (e.g. if it's worth logging)

Reply:
yeah, valid - done!